### PR TITLE
Update Schema: Prompt becomes a table and add subprompt table

### PR DESCRIPTION
**Pull request by andrew-rietz**
- Prompt has become its own table (previously in flashcard)
- Added sub-prompt table

Issues:
- No option to create a user-defined type in the DBDesigner.net interface. Currently using json as the type for both the "prompt" and "sub_prompt" fields within the [prompt] and [sub-prompt] tables respectively.

Addresses #17 and #18

### CODE REVIEW

**Comment by andrew-rietz** [comment]
Taylor, it looks like my solution for generating the ERD and schema file may not work out after all. Kind of a bummer... Take a look and let's circle back. I wasn't able to create a user-defined type ("multiple_choice") but opted to use a json data type instead. I think it's workable, but may require rethinking how we handle input and database I/O.

File: schema/schema_updated_17.sql [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	CONSTRAINT "flashcard_tag_pk" PRIMARY KEY ("flashcard_id","tag_id")
+) WITH (
+  OIDS=FALSE
+);
+
+
+
+CREATE TABLE "prompt" (
+	"id" serial NOT NULL DEFAULT 'uuid_generate_v1()',
+	"type" json NOT NULL,
```
Comment (tmrietz) [inline schema/schema_updated_17.sql]: I think we can make "type" a varchar(30) type and add a check that enforces 2 types right now: "text" and "multiple_choice"

**Comment by tmrietz** [comment]
Can we limit the whitespace a bit? And do we need to include the binary file? Other than that and comment on the "type" field's type, looks good.

**Comment by andrew-rietz** [comment]
Yes, we can limit the whitespace. I had been using the DBDesigner.net site to create the ERD and automatically generate the underlying sql code, and didn't do any cleaning after exporting. It sounds like we have needs that are beyond the scope of what DBDesigner offers (checks, for instance), so maybe it will become a moot point.

The binary file is an image of the ERD. If we're moving away from the DBDesigner solution, maybe this is an option: https://github.com/BurntSushi/erd/blob/master/README.md

If we go with the above, we could probably put together a quick script that would take our text representation of the schema and create a sql file from it.

File: schema/schema_updated_17.sql [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	"prompt" TEXT NOT NULL UNIQUE,
+	CONSTRAINT "prompt_pk" PRIMARY KEY ("id")
+) WITH (
+  OIDS=FALSE
+);
+
+
+
+CREATE TABLE "sub-prompt" (
```
Comment (tmrietz) [inline schema/schema_updated_17.sql]: I missed this but "sub_prompt" for consistency instead of "sub-prompt"

**Comment by tmrietz** [comment]
Ok got it. I think for now let's continue to consider options for the ERD but keep this PR limited to actual schema changes and defining those in new and existing *.sql files in the schema/sql directory. 

You've obviously made most of the changes already on your local DB so just have to make a couple tweaks and write 'em in .sql files for now. The tweaks will address the comments about the "type" field and the "sub-prompt"-> "sub_prompt" change. There's an example at [https://www.postgresql.org/docs/9.4/ddl-constraints.html](url) section 5.3.1 that shows how to add multiple checks for a single field.

File: schema/schema_updated_17.sql (hunk 1 above)
Comment (andrew-rietz) [reply schema/schema_updated_17.sql]: I get it now... I thought you wanted some type of custom data type. Really you want the text values within "type" to be either "text" or "multiple_choice".


This should be updated and incorporated in latest commit

File: schema/schema_updated_17.sql (hunk 2 above)
Comment (andrew-rietz) [reply schema/schema_updated_17.sql]: Updated

**Comment by andrew-rietz** [comment]
Moved away from the single sql file approach. Split each table out into its own file and adopted the same syntax / styling you used originally because it's less verbose and easier to follow.

**Pull request merged**